presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -303,9 +303,9 @@ private void rehash()
         this.valuesByGroupId.ensureCapacity(maxFill);
     }
 
-    private static int getHashPosition(long rawHash, int mask)
+    private static long getHashPosition(long rawHash, int mask)
     {
-        return ((int) murmurHash3(rawHash)) & mask;
+        return murmurHash3(rawHash) & mask;
```

Comment:
Are you sure about that?
It looks like it returns 64 bit hashes:
https://ideone.com/ykpZ47 (murmurHash3 copied from decompiled it.unimi.dsi.fastutil.HashCommon class file).